kt/godot-runtime/src/main/kotlin/godot/core/Variants.kt

```diff
@@ -232,11 +233,19 @@ class KtVariant {
 
     inline fun <reified T : KtObject> asObject(): T {
         val objectValue = data.objectValue
+        val ptr = objectValue.ptr
         val constructorIndex = objectValue.engineConstructorIndex
-        return KtObject.instantiateWith(
-                objectValue.ptr,
+        val isRef = objectValue.isRef
+        return (if (isRef) {
+            GarbageCollector.refWrappedMap[ptr]?.get()
```

Comment:
we need to synchronize this right? `refWrappedMap` can be modified by the another thread at this point.

Reply:
Hum correct.  This current implementation is dangerous.

Reply:
Yep I agree. Maybe we should go for queue now @CedNaru